core/src/main/java/io/opencensus/tags/TagContext.java

```diff
@@ -56,16 +58,28 @@ public boolean equals(Object other) {
     if (!(other instanceof TagContext)) {
       return false;
     }
-    HashMultiset<Tag> tags1 = HashMultiset.create(Lists.newArrayList(unsafeGetIterator()));
-    HashMultiset<Tag> tags2 =
-        HashMultiset.create(Lists.newArrayList(((TagContext) other).unsafeGetIterator()));
+    TagContext otherTags = (TagContext) other;
+    Iterator<Tag> iter1 = unsafeGetIterator();
+    Iterator<Tag> iter2 = otherTags.unsafeGetIterator();
+    Multiset<Tag> tags1 =
+        iter1 == null
+            ? ImmutableMultiset.<Tag>of()
+            : HashMultiset.create(Lists.newArrayList(iter1));
+    Multiset<Tag> tags2 =
+        iter2 == null
+            ? ImmutableMultiset.<Tag>of()
+            : HashMultiset.create(Lists.newArrayList(iter2));
     return tags1.equals(tags2);
   }
 
   @Override
   public final int hashCode() {
     int hashCode = 0;
-    for (Iterator<Tag> i = unsafeGetIterator(); i.hasNext(); ) {
```

Comment:
Should we use something like this: http://docs.oracle.com/javase/7/docs/api/java/util/Arrays.html?is-external=true#hashCode(java.lang.Object[]) or https://google.github.io/guava/releases/19.0/api/docs/com/google/common/base/Objects.html#hashCode(java.lang.Object...)

Reply:
We need to ensure that the hash code is order independent, so I think it is better to explicitly sum the elements' hash codes.